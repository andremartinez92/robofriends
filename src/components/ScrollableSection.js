// @flow

import React from 'react';

const styles = {
  overflow: 'scroll',
  height: '800px',
};

type Props = {
  children: React$Element<*>,
};

function ScrollableSection(props: Props) {
  const { children } = props;
  return <div style={styles}>{children}</div>;
}

export default ScrollableSection;
