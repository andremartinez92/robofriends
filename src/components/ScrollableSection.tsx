import React from 'react';

const styles = {
  overflow: 'scroll',
  height: '800px',
};

interface Props {
  children: React.ReactNode;
}

function ScrollableSection(props: Props): JSX.Element {
  const { children } = props;
  return <div style={styles}>{children}</div>;
}

export default ScrollableSection;
