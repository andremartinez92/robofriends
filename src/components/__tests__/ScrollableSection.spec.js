import React from 'react';
import { shallow } from 'enzyme';

import ScrollableSection from '../ScrollableSection';

describe('<ScrollableSection>', () => {
  describe('when has no children', () => {
    it('renders the expected component', () => {
      const subject = shallow(<ScrollableSection />);
      expect(subject).toMatchSnapshot();
    });
  });

  describe('when has one child', () => {
    it('renders the expected component', () => {
      const subject = shallow(
        <ScrollableSection>
          <div>Hello</div>
        </ScrollableSection>
      );
      expect(subject).toMatchSnapshot();
    });
  });

  describe('when has multiple children', () => {
    it('renders the expected component', () => {
      const subject = shallow(
        <ScrollableSection>
          <div>Hello</div>
          <div>World</div>
        </ScrollableSection>
      );
      expect(subject).toMatchSnapshot();
    });
  });
});
