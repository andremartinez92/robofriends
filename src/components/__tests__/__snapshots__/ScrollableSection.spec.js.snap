// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollableSection> when has multiple children renders the expected component 1`] = `
<div
  style={
    Object {
      "height": "800px",
      "overflow": "scroll",
    }
  }
>
  <div>
    Hello
  </div>
  <div>
    World
  </div>
</div>
`;

exports[`<ScrollableSection> when has no children renders the expected component 1`] = `
<div
  style={
    Object {
      "height": "800px",
      "overflow": "scroll",
    }
  }
/>
`;

exports[`<ScrollableSection> when has one child renders the expected component 1`] = `
<div
  style={
    Object {
      "height": "800px",
      "overflow": "scroll",
    }
  }
>
  <div>
    Hello
  </div>
</div>
`;
